java/compaction/compaction-core/src/test/java/sleeper/compaction/core/job/dispatch/CompactionJobDispatcherTest.java

```diff
@@ -272,6 +272,27 @@ void shouldFailForRetryWhenOneJobFailedToSend() {
         assertThat(pendingDeadLetterQueue).isEmpty();
     }
 
+    @Test
+    void shouldDeleteAllFilesAfterBatchIsSent() {
+
+        // Given
+        FileReference file1 = fileFactory.rootFile("test1.parquet", 1234);
+        CompactionJob job1 = compactionFactory.createCompactionJob("test-job-1", List.of(file1), "root");
+        update(stateStore).addFiles(List.of(file1));
+        assignJobIds(List.of(job1));
+
+        CompactionJobDispatchRequest request = generateBatchRequestAtTime(
+                "test-batch", Instant.parse("2024-11-15T10:30:00Z"));
+        putCompactionJobBatch(request, List.of(job1));
+
+        // When
+        dispatchWithNoRetry(request);
+
+        // Then
+        assertThat(compactionQueue).containsExactly(job1);
+        assertThat(s3PathToCompactionJobBatch).isEmpty();
+    }
+
```

Comment:
I think we need to handle the failure case. Right now if the deletion fails, the whole batch of compaction jobs will be put back on the queue and sent again. They'll all be run twice, or as many times as the deletion fails until it ends up on the dead letter queue. That could be quite expensive.